### ISSUE TITLE
Daemon should verify authz plugins on startup

### ISSUE THREAD

**Issue by thaJeztah** [docker/docker#27909]
This is a tracking issue for a discussion I was having with @anusha-ragunathan about https://github.com/docker/docker/pull/27804.

Currently, if a uses specifies a non-existing authz plugin, the daemon will still start, but the first API request trying to call the authz plugin will fail (due to "lazy loading" plugins).

Specifying a non-existent authz plugin should be verified during startup, and the daemon should fail to start, otherwise a user may _think_ the authz plugin is enabled, but actually not.

/cc @riyazdf

**Comment by thaJeztah** [comment]
ping @runcom @liron-l perhaps you have some thoughts on this? :smile:

**Comment by anusha-ragunathan** [comment]
Spoke to @riyazdf and he will be working on this.

**Comment by liron-l** [comment]
Total agree @thaJeztah, daemon should stop if plugin is invalid.
I also think #27804 should be reverted. I find it a little insecure that a plugin is blacklisted while the daemon is running.

**Comment by anusha-ragunathan** [comment]
@liron-l : Without https://github.com/docker/docker/pull/27804, you will crash the daemon when started with an plugin that's not available. The problem is, there's NO validation performed on the validity of these authzplugins on daemon start. Currently, the first API request is the one triggering the load of the authzplugin and hence it has to when the daemon is running.

**Comment by liron-l** [comment]
I meant to remove #27804 once we check that the authz plugin is valid on daemon startup (and crash otherwise).
The reasoning is that configuring authz probably means that the TCP endpoint is used (since for now, TLS is the only authn mechanism available), and removing the plugin makes the endpoint completely insecure.

**Comment by anusha-ragunathan** [comment]
sure. once that `check plugin is valid on startup` PR is merged, we can remove this check. btw, we should never crash the daemon. handle the error gracefully and exit.

**Comment by thaJeztah** [comment]
@riyazdf @liron-l can one of you take care of removing https://github.com/docker/docker/pull/27804 if that's no longer needed?

**Comment by riyazdf** [comment]
@thaJeztah: sure, I can handle it
